libs/ui/packages/native/src/components/Layout/List/VerticalTimeline/TimelineIndicator.tsx

```diff
@@ -1,30 +1,84 @@
 import React from "react";
+import { Dimensions, PixelRatio, Platform, StyleSheet } from "react-native";
 import { CircledCheckSolidMedium } from "@ledgerhq/icons-ui/native";
 import styled, { useTheme } from "styled-components/native";
+import Svg, { Line } from "react-native-svg";
 
 import Flex, { Props as FlexProps } from "../../Flex";
 import { ItemStatus } from "../types";
 import { Theme } from "src/styles/theme";
 
-const TopSegment = styled(Flex)<{ status: ItemStatus; hidden?: boolean }>`
-  height: 20px;
-  border-width: ${(p) => (p.hidden ? 0 : 1)}px;
-  border-style: solid;
-  border-color: ${(p) =>
-    p.status === "inactive" ? p.theme.colors.neutral.c50 : p.theme.colors.primary.c80};
-  background: ${(p) =>
-    p.status === "inactive" ? p.theme.colors.neutral.c50 : p.theme.colors.primary.c80};
-`;
+type SegmentProps = { status: ItemStatus; hidden?: boolean; height?: number };
 
-const BottomSegment = styled(Flex)<{ status: ItemStatus; hidden?: boolean }>`
-  flex: 1;
-  border-width: ${(p) => (p.hidden ? 0 : 1)}px;
-  border-style: solid;
-  border-color: ${(p) =>
-    p.status === "completed" ? p.theme.colors.primary.c80 : p.theme.colors.neutral.c50};
-  background: ${(p) =>
-    p.status === "completed" ? p.theme.colors.primary.c80 : p.theme.colors.neutral.c50};
-`;
+const dashLength = 4;
+const linesWidth = 2;
+
+/**
+ * We have to set a fixed size for the lines (and then they get clipped with overflow:hidden)
+ * because their real height is dynamic (depends on the height of the step) and
+ * there seems to be no way to have that height being dynamic without having the
+ * scaling effect affecting the length of the stroke dash (the "- - - ").
+ * This is kind of a trick but seemingly the only way to have a nice stroke dash
+ **/
+const linesLength = 5000;
+
+const TopSegmentSvg: React.FC<SegmentProps> = ({ status, hidden, height }) => {
+  const theme = useTheme();
+  const strokeColor = status === "inactive" ? theme.colors.neutral.c50 : theme.colors.primary.c80;
+  const strokeDashArray = status === "inactive" ? `${dashLength}` : undefined;
+  return (
+    <Flex height={height} width={"100%"} overflow="hidden">
+      <Flex style={StyleSheet.absoluteFillObject} alignItems="center">
+        <Svg
+          height={linesLength}
+          width={linesWidth}
+          viewBox={`0 0 ${linesWidth} ${linesLength}`}
+          preserveAspectRatio="xMinYMin slice"
+        >
+          <Line
+            x1="0"
+            y1="0"
+            x2="0"
+            y2="100"
+            strokeWidth={hidden ? 0 : 2 * linesWidth}
+            stroke={strokeColor}
+            strokeDasharray={strokeDashArray}
+          />
+        </Svg>
```

Comment:
Doing some magic here with SVG and all that i'm not sure to fully understand, but it looks good 👌

Reply:
Yeah don't worry I don't fully understand it either 😆 
The thing is that this component always has the same height (`linesLength`), but then when we display it we "clip" it. That constant height is necessary to avoid weird stretching effects, so that all TopSegment and BottomSegment instances have dashes of the same height, without stretching due to the SVG's height. That "clipping" is done with a container that has a dynamic height and absolute positioning, its children having a constant height.

Then other things that I'm doing in this file is that to join properly the "top segment" and "bottom segment":
- the "BottomSegment" is flipped vertically, so it's "starts" at the bottom and goes it with this pattern
  "- - - - -" (it starts with a dash)
- the "TopSegment" is not flipped, so it's "starts" at the top and goes to the bottom, with this pattern
  " - - - - " (it starts with an empty space)